subsys/bluetooth/audio/shell/bap.c

```diff
@@ -2502,24 +2680,96 @@ static void stream_configured_cb(struct bt_bap_stream *stream,
 			sh_stream->lc3_frame_duration_us = 0U;
 		}
 
-		ret = bt_audio_codec_cfg_get_frame_blocks_per_sdu(stream->codec_cfg, true);
-		if (ret > 0) {
-			sh_stream->lc3_frames_per_sdu = (uint8_t)ret;
+		ret = bt_audio_codec_cfg_get_chan_allocation(codec_cfg,
+							     &sh_stream->lc3_chan_allocation);
+		if (ret == 0) {
+			sh_stream->lc3_chan_cnt = get_chan_cnt(sh_stream->lc3_chan_allocation);
+		} else {
+			shell_error(ctx_shell, "Could not get channel allocation: %d", ret);
+			sh_stream->lc3_chan_allocation = BT_AUDIO_LOCATION_MONO_AUDIO;
+			sh_stream->lc3_chan_cnt = 1U;
+		}
+
+		ret = bt_audio_codec_cfg_get_frame_blocks_per_sdu(codec_cfg, true);
+		if (ret >= 0) {
+			sh_stream->lc3_frame_blocks_per_sdu = (uint8_t)ret;
 		} else {
 			shell_error(ctx_shell, "Could not get frame blocks per SDU: %d", ret);
-			sh_stream->lc3_frames_per_sdu = 0U;
+			sh_stream->lc3_frame_blocks_per_sdu = 0U;
 		}
 
-		ret = bt_audio_codec_cfg_get_octets_per_frame(stream->codec_cfg);
-		if (ret > 0) {
+		ret = bt_audio_codec_cfg_get_octets_per_frame(codec_cfg);
+		if (ret >= 0) {
 			sh_stream->lc3_octets_per_frame = (uint16_t)ret;
 		} else {
 			shell_error(ctx_shell, "Could not get octets per frame: %d", ret);
 			sh_stream->lc3_octets_per_frame = 0U;
 		}
+
+#if defined(CONFIG_BT_AUDIO_RX)
+		if (info.can_recv) {
+			if (sh_stream->lc3_decoder == NULL) {
+				const int err = init_lc3_decoder(sh_stream);
+
+				if (err != 0) {
+					shell_error(ctx_shell, "Failed to init the LC3 decoder: %d",
+						    err);
+
+					return;
+				}
+			}
+
+			sh_stream->decoded_cnt = 0U;
+#endif /* CONFIG_BT_AUDIO_RX */
+		}
 	}
 #endif /* CONFIG_LIBLC3 */
 
+#if defined(CONFIG_BT_AUDIO_TX)
+	sh_stream->connected_at_ticks = k_uptime_ticks();
+#endif /* CONFIG_BT_AUDIO_TX */
+
+#if defined(CONFIG_BT_AUDIO_RX)
+	sh_stream->lost_pkts = 0U;
+	sh_stream->err_pkts = 0U;
+	sh_stream->dup_psn = 0U;
+	sh_stream->rx_cnt = 0U;
+	sh_stream->dup_ts = 0U;
+#endif
+}
+
+static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
+{
+	printk("Stream %p stopped with reason 0x%02X\n", stream, reason);
+
+#if defined(CONFIG_LIBLC3)
+	lc3_clear_stream(shell_stream_from_bap_stream(stream));
+#endif /* CONFIG_LIBLC3 */
+
+#if defined(CONFIG_BT_BAP_BROADCAST_SINK)
+	struct shell_stream *sh_stream = shell_stream_from_bap_stream(stream);
+
+	if (IS_ARRAY_ELEMENT(broadcast_sink_streams, sh_stream)) {
+		if (default_broadcast_sink.stream_cnt != 0) {
+			default_broadcast_sink.stream_cnt--;
+		}
+
+		if (default_broadcast_sink.stream_cnt == 0) {
+			/* All streams in the broadcast sink has been terminated */
+			default_broadcast_sink.syncable = true;
```

Comment:
why do we need to set default_broadcast_sink.syncable = true ?

Reply:
Yeah that doesn't make any sense, especially since we are setting it to `false` just below. 

Must have been a bad merge conflict resolution or something :o